app/modules/intelligence/tools/web_tools/github_tool.py

```
@@ -126,19 +140,27 @@ def _get_github_client(self, repo_name: str) -> Github:
                 return self.get_public_github_instance()
             except Exception as public_error:
                 logging.error(f"Failed to access public repo: {str(public_error)}")
-                raise Exception(f"Repository {repo_name} not found or inaccessible on GitHub")
+                raise Exception(
+                    f"Repository {repo_name} not found or inaccessible on GitHub"
+                )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)